database/gredis/gredis_redis_group_generic.go

```diff
@@ -60,3 +61,9 @@ type ExpireOption struct {
 	GT bool // GT -- Set expiry only when the new expiry is greater than current one
 	LT bool // LT -- Set expiry only when the new expiry is less than current one
 }
+
+type ScanOption struct {
+	Match *string // Match -- Specifies a glob-style pattern for filtering keys.
+	Count *int    // Count -- Suggests the number of keys to return per scan.
+	Type  *string // Type -- Filters keys by their data type. Valid types are "string", "list", "set", "zset", "hash", and "stream".
+}
```

Comment:
- Missing comments for public struct. 
_You might say there're no comments for some of the public interface methods and types either, but you see this is heavy work that we have not finished yet._
- Is it necessary using pointer as the option attribute? It's not convenient for developers to use it, as the developers would need convert the variable value to pointer before using this option.

Reply:
- I updated the comments for the `ScanOption` public struct. 
- About using pointer as the option attributes: Because if we don't use pointer, the `convertOptionToArgs` function will appends attributes with their zero value into `args` variable. The `Do` method will executes `Scan` command with all of values in the `args` and returns unexpected result. 

For example:
- Data in Redis:
```json
"firstname": "Jack",
"lastname":  "Stuntman",
"age":       35,
"nickname":  "Jumper"
```

- Run Test:
```Go
type ScanOption struct {
  Match string 
  Count int
  Type  string
}

func Test_GroupGeneric_Scan(t *testing.T) {
    gtest.C(t, func(t *gtest.T) {
        performScan := func(cursor uint64, option *ScanOption) ([]string, error) {
	        var allKeys = []string{}
	        for {
		        var nextCursor uint64
		        var keys []string
		        var err error
        
		        if option != nil {
			        nextCursor, keys, err = redis.Scan(ctx, cursor, *option)
		        } else {
			        nextCursor, keys, err = redis.Scan(ctx, cursor)
		        }
		        if err != nil {
			        return nil, err
		        }
        
		        allKeys = append(allKeys, keys...)
		        if nextCursor == 0 {
			        break
		        }
		        cursor = nextCursor
	        }
	        return allKeys, nil
        }
        
        optWithName := gredis.ScanOption{Match: "*name*", Count: 10}
        keysWithName, err := performScan(0, &optWithName)
        t.AssertNil(err)
        if len(keysWithName) < 3 {
	        t.Errorf("Expected at least 3 keys matching '*name*', found %d", len(keysWithName))
        }
    }
}
```

- Test result: 
```
Expected at least 3 keys matching '*name*', found 0
```

That means the `Scan` method also accepts the field `Type` with empty string instead of ignores it. So if we want to the `MATCH/COUNT/TYPE` is optional, I have to define those fields's data type is pointer. 

If you have the other solution, I really looking forward to learn from it.

Reply:
@phuonganhniie You can macnually convert the option to the args that will be passed to redis, for example:
```go
var args = make([]any, 0)
if usedOption.Match != "" {
    args = append(args, "MATCH", usedOption, Match)
}
if usedOption.Count > 0 {
    args = append(args, "COUNT", usedOption, Count)
}
if usedOption.Type != "" {
    args = append(args, "TYPE", usedOption, Type)
}
v, err := r.Operation.Do(ctx, "Scan", args...)
```

Or else, you can manually convert the public `ScanOption` to internal `doScanOption` using pointer attributes, which can be directly used in function `mustMergeOptionToArgs`, for example:
```go
type ScanOption struct {
	Match string // Match -- Specifies a glob-style pattern for filtering keys.
	Count int    // Count -- Suggests the number of keys to return per scan.
	Type  string // Type -- Filters keys by their data type. Valid types are "string", "list", "set", "zset", "hash", and "stream".
}

type doScanOption struct {
	Match *string 
	Count *int   
	Type  *string 
}
```

Reply:
@gqcn Hi John, I've updated by transform the public `ScanOption` to internal `doScanOption` using pointer attributes. Please help me review it and thank you for your patience.